### PR TITLE
use croaring serialization functions

### PR DESCRIPTION
**Pull request by yuzhichang**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Performance Improvement

Short description (up to few sentences):
Use CRoaring serialization functions instead of home-brewed ones.
This change improved serialization/deserialization by 2x faster, and reduced bitmap disk usage to about 50%.

WARNING: This change breaks backwards compatibility. However I think few people are using clickhouse bitmap column.

```
Query 1:
SELECT bitmapCardinality((SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_market_NA'))
Query 2:
SELECT count(*) FROM (SELECT bitmapCardinality((SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_market_NA')))
Query 3:
SELECT count(*) FROM(SELECT bitmapAnd((SELECT mid_seqs FROM cdp_tags WHERE tag_id='member_origin_channel_NA'),(SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_province_广东省')))
Query 4:
SELECT bitmapCardinality(bitmapAnd((SELECT mid_seqs FROM cdp_tags WHERE tag_id='member_origin_channel_NA'),(SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_province_广东省')))
Query 5:
SELECT bitmapCardinality(bitmapAnd((SELECT mid_seqs FROM cdp_tags WHERE tag_id='member_origin_channel_NA'),bitmapBuild([1,2,3,12345678])))
Query 6:
SELECT bitmapAndCardinality(bitmapAnd(bitmapAnd((SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_market_NA'), (SELECT mid_seqs FROM cdp_tags WHERE tag_id='first_buy_province_广东省')), (SELECT mid_seqs FROM cdp_tags WHERE tag_id='member_origin_channel_NA')), (SELECT mid_seqs FROM cdp_tags WHERE tag_id='mobile_province_NA') );


Before vs After
Query 1: 1.27s vs 0.57s
Query 2: 2.1s vs 0.77s
Query 3: 15.2s vs 6.55s
Query 4: 9.7s vs 4.45s
Query 5: 7.667s vs 3.78s
Query 6: 22.55s vs 8.57s
```

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Do we need to mention backwards incompatibility (if someone was storing aggregation states on disk)?

File: dbms/src/AggregateFunctions/AggregateFunctionGroupBitmapData.h [hunk 1]
```diff
@@ -103,9 +104,10 @@ class RoaringBitmapWithSmallSet : private boost::noncopyable
 
         if (isLarge())
         {
-            UInt32 cardinality = roaring_bitmap_get_cardinality(rb);
-            writePODBinary(cardinality, out);
-            db_ra_to_uint32_array(out, &rb->high_low_container);
+            uint32_t expectedsize = roaring_bitmap_portable_size_in_bytes(rb);
+            std::string s(expectedsize,0);
+            roaring_bitmap_portable_serialize(rb, const_cast<char*>(s.data()));
+            writeStringBinary(s,out);
```
Comment (filimonov) [inline dbms/src/AggregateFunctions/AggregateFunctionGroupBitmapData.h]: Is that change backwards compatible? I.e. data written by new variant of code can be read by old variant and vice versa?

Comment (alexey-milovidov) [reply dbms/src/AggregateFunctions/AggregateFunctionGroupBitmapData.h]: It's mentioned in PR description.

**Comment by nicelulu** [comment]
However, older versions of the table column with bitmap will not work properly.

**Pull request merged**